### PR TITLE
IPMVAL-3357 fix

### PR DESCRIPTION
**Pull request by FrancoisRegisDegott-eaton**
due to filesystem migration from cxxtools to stdlib

### CODE REVIEW

**Review by stark-dev — APPROVED** [review]

**Review by perrettecl — APPROVED** [review]

File: src/linuxmetric.cc [hunk 1]
```diff
@@ -450,7 +450,12 @@ zhashx_t* linuxmetric_list_interfaces(const std::string& root_dir)
     std::filesystem::path dir(root_dir + "sys/class/net/");
 
     for (const auto & entry : std::filesystem::directory_iterator(dir)) {
-        std::string iface = std::filesystem::relative(entry.path(), dir);
+        std::string iface = entry.path();
+        auto pos = iface.find_last_of("/");
+        if (pos != std::string::npos) {
+            iface = iface.substr(pos + 1);
+        }
+
```
Comment (etnKlendathu) [inline src/linuxmetric.cc:458]: Just a question: this has different result than relative dir as it was before?

File: src/fty_info_server.cc [hunk 2]
```diff
@@ -238,24 +239,35 @@ static void s_publish_announce(fty_info_server_t* self)
...
 
+    int ttl = 3 * self->linuxmetrics_interval; // in seconds
     linuxmetric_t* metric = static_cast<linuxmetric_t*>(zlistx_first(info));
     while (metric) {
         char* value = zsys_sprintf("%lf", metric->value);
         log_debug("Publishing metric %s, value %lf, unit %s", metric->type, metric->value, metric->unit);
 
-        if (fty::shm::write_metric(rc_iname, metric->type, value, metric->unit, ttl) == 0) {
+        int r = fty::shm::write_metric(rc_iname, metric->type, value, metric->unit, ttl);
+        if (r == 0) {
```
Comment (etnKlendathu) [inline src/fty_info_server.cc:264]: in c++17 you could scope of `if` statement...
```cpp
if (int r = fty::shm::write_metric(rc_iname, metric->type, value, metric->unit, ttl); r == 0) {
```
so, `r` would be available only in this `if` branches, not in the whole scope

File: src/fty_info_server.cc [hunk 3]
```diff
@@ -238,24 +239,35 @@ static void s_publish_announce(fty_info_server_t* self)
 //  publish Linux system info on STREAM METRICS
 static void s_publish_linuxmetrics(fty_info_server_t* self)
 {
-    log_debug("s_publish_linuxmetrics");
+    char* rc_iname = topologyresolver_id(self->resolver);
```
Comment (etnKlendathu) [inline src/fty_info_server.cc:242]: Just a question: rc_iname should be freed?

File: src/linuxmetric.cc (hunk 1 above)
Comment (FrancoisRegisDegott-eaton) [reply src/linuxmetric.cc:458]: Don't know if it is really a cxxtools->stdlib problem...
but if files are symlinks.. and relativepath() isn't efficient on such case...

**Pull request merged**